x/concentrated-liquidity/position_test.go

```diff
@@ -179,7 +210,113 @@ func (s *KeeperTestSuite) TestGetPosition() {
 				s.Require().NoError(err)
 				s.Require().Equal(test.expectedPosition, position)
 			}
-
 		})
 	}
 }
+
+// Thorough fuzz testing for join-time sumtrees
+func (s *KeeperTestSuite) TestMultiplePositionsFuzz() {
+	// Init suite for fuzz testing
+	s.Setup()
+
+	// Create a default CL pool
+	s.PrepareConcentratedPool()
+
+	// Test 100 random positions in same pool at random jointimes
+	// Note: we can fuzz these values if needed as well
+	numPositions := 100
+	validPoolId := uint64(1)
+
+	// Max lower and upper ticks
+	tickLowerBound := int64(100)
+	tickUpperBound := int64(100)
+
+	// Max time elapsed between positions (in seconds)
+	timeElapsedUpperBound := 1000
+
+	// We generate a new test account per position for robustness
+	testAccounts := osmoutils.CreateRandomAccounts(numPositions)
+
+	// We use the reference time to check if our sumtree properly accumulates liquidity
+	// Max seconds set at 32 bits & max nanoseconds at 30 bits to comply with bounds
+	// specified in the Time package
+	referenceTime := time.Unix(int64(rand.Int31()), int64(rand.Int31())%(1<<30))
+	maxElapsedTimeFromStart := timeElapsedUpperBound * numPositions
+
+	// Set start time = ref time - one tenth of max time elapsed
+	// This should mean we end up with at least some liquidity on each side of the
+	// ref time, ensuring testability while still having some variance in join times
+	startTime := referenceTime.Add(time.Duration(-1*maxElapsedTimeFromStart/10) * time.Second)
+	s.Ctx = s.Ctx.WithBlockTime(startTime)
+
+	// Trackers for total liquidity on either side of our reference join time
+	expLiqBeforeOrAtRefTime := sdk.ZeroDec()
+	expLiqAfterRefTime := sdk.ZeroDec()
+
+	for i := 0; i < numPositions; i++ {
+		// Generate random time elapsed since last join and update block time
+		timeElapsed := time.Duration(rand.Int()%timeElapsedUpperBound) * time.Second
+
+		// Half the time, override time elapsed and add to the same join time as the previous position (time elapsed = 0)
+		timeElapsed = timeElapsed * time.Duration(rand.Int()%2)
+
+		// Update current block time to reflect time elapsed
+		joinTime := s.Ctx.BlockTime().Add(timeElapsed)
+		s.Ctx = s.Ctx.WithBlockTime(joinTime)
+
+		// Generate random lower and upper ticks
+		lowerTick := (-1) * (rand.Int63() % tickLowerBound)
+		upperTick := rand.Int63() % tickUpperBound
```

Comment:
I like what you did with this fuzz testing, but I think we might need to expand this to prevent every lower tick from being negative and every upper tick from being positive. Seems like we might be leaving some edge cases untested.